File: videojuegoDeChinpokomon/src/logger/Logger.java
package logger;

public class Logger {

    private static Logger instance;

    private static Level levelDefault = Level.DEBUG;

    public static void setLevel(Level level){
        Logger.levelDefault = level;
    }

    public static Logger getInstance(){
        if (instance == null) {
            instance = new Logger(levelDefault);
        }
        return instance;
    }

    private Level level;

    public void setLogLevel(Level level) {
        this.level = level;
    }

    private Logger(Level level){
        this.level = level;
    }

    public void debug(String message) {
        this.printMessage(level.DEBUG, message);
    }

    public void info(String message) {
        this.printMessage(level.INFO, message);;
    }

    public void warn(String message) {
        this.printMessage(level.WARN, message);;
    }

    public void error(String message) {
        this.printMessage(level.ERROR, message);;
    }

    public void printMessage(Level level, String msg) {
        if (this.level.level <= level.level) {
            System.out.println(msg);
        }
    }

}

File: ChimpokomonPython/ModuloSRC/Logger/Logger.py
from Logger.Level import Level

class Logger():
    
    _instance = None

    def __init__(self):
        None
    
    @classmethod
    def instance(cls, level=Level.DEBUG):
        if cls._instance is None:
            cls._instance = cls.__new__(cls)
            cls._level = level
        return cls._instance

    @property
    def level(self):
        return self._level

    @level.setter
    def level(self, level):
        self._level = level

    def debug(self, respuesta):
        print("DEBUG: " + respuesta)

    def info(self, respuesta):
        if(self.level != Level.DEBUG):
            print("INFO: " + respuesta)

    def warn(self, respuesta):
        if( self.level != Level.DEBUG or self.level == Level.INFO):
            print("WARN: " + respuesta)

    def error(self, respuesta):
        if( self.level == Level.ERROR):
            print("ERROR: " + respuesta)